plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -296,4 +292,12 @@ public static Object deserializePartitionValue(Type type, String valueString, St
 
         return Collections.unmodifiableMap(partitionKeys);
     }
+
+    public static LocationProvider getLocationProvider(String tableLocation, Map<String, String> properties)
```

Comment:
@alexjo2144 I think this is the best we can do on Trino side, the 2 location provider are protected and not public classes, so we can only block the creation if `locaiton-impl` table property is set.